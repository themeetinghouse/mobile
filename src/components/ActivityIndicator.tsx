import React from 'react';
import {
  View,
  ActivityIndicator as NativeActivityIndicator,
  StyleSheet,
  ViewStyle,
} from 'react-native';
import Theme from '../Theme.style';

const style = StyleSheet.create({
  container: {
    padding: 15,
  },
});

interface Props {
  animating?: boolean;
  style?: ViewStyle;
}

export default function ActivityIndicator(props: Props): JSX.Element {
  return (
    <View style={[style.container, props.style]}>
      <NativeActivityIndicator
        size="large"
        color={Theme.colors.white}
        animating={props.animating}
      ></NativeActivityIndicator>
    </View>
  );
}
