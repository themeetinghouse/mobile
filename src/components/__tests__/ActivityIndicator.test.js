import React from 'react';
import { render } from '@testing-library/react-native';
import ActivityIndicator from '../ActivityIndicator';

test('Active indicator', () => {
  const { toJSON } = render(<ActivityIndicator animating />);

  expect(toJSON()).toMatchSnapshot();
});

test('Inactive indicator', () => {
  const { toJSON } = render(<ActivityIndicator animating={false} />);

  expect(toJSON()).toMatchSnapshot();
});
