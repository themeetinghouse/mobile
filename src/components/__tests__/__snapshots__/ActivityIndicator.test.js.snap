// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Active indicator 1`] = `
<View
  style={
    Array [
      Object {
        "padding": 15,
      },
      undefined,
    ]
  }
>
  <ActivityIndicator
    animating={true}
    color="#ffffff"
    hidesWhenStopped={true}
    size="large"
  />
</View>
`;

exports[`Inactive indicator 1`] = `
<View
  style={
    Array [
      Object {
        "padding": 15,
      },
      undefined,
    ]
  }
>
  <ActivityIndicator
    animating={false}
    color="#ffffff"
    hidesWhenStopped={true}
    size="large"
  />
</View>
`;
